libnavigation-ui/src/main/java/com/mapbox/navigation/ui/instruction/NavigationAlertView.java

```diff
@@ -0,0 +1,154 @@
+import static com.mapbox.navigation.ui.legacy.NavigationConstants.FEEDBACK_BOTTOM_SHEET_DURATION;
+
+public class NavigationAlertView extends AlertView implements FeedbackBottomSheetListener {
+
+  private static final long THREE_SECOND_DELAY_IN_MILLIS = 3000;
+  private NavigationViewModel navigationViewModel;
+  private boolean isEnabled = true;
+
+  public NavigationAlertView(Context context) {
+    this(context, null);
+  }
+
+  public NavigationAlertView(Context context, @Nullable AttributeSet attrs) {
+    this(context, attrs, -1);
+  }
+
+  public NavigationAlertView(Context context, @Nullable AttributeSet attrs, int defStyleAttr) {
+    super(context, attrs, defStyleAttr);
+  }
+
+  /**
+   * Sets the NavigationViewModel in the view
+   *
+   * @param navigationViewModel to set
+   */
+  public void subscribe(NavigationViewModel navigationViewModel) {
+    this.navigationViewModel = navigationViewModel;
+  }
+
+  /**
+   * Shows this alert view for when feedback is submitted
+   */
+  public void showFeedbackSubmitted() {
+    if (!isEnabled) {
+      return;
+    }
+    show(getContext().getString(R.string.feedback_submitted), THREE_SECOND_DELAY_IN_MILLIS, false);
+  }
+
+  /**
+   * Shows this alert view to let user report a problem for the given number of milliseconds
+   */
+  public void showReportProblem() {
+    if (!isEnabled) {
+      return;
+    }
+    final Handler handler = new Handler();
+    handler.postDelayed(new Runnable() {
+      @Override
+      public void run() {
+        show(getContext().getString(R.string.report_problem),
+                ALERT_VIEW_PROBLEM_DURATION, true);
+      }
+    }, THREE_SECOND_DELAY_IN_MILLIS);
+  }
+
+  /**
+   * Shows {@link FeedbackBottomSheet} and adds a listener so
+   * the proper feedback information is collected or the user dismisses the UI.
+   */
+  public void showFeedbackBottomSheet() {
+    if (!isEnabled) {
+      return;
+    }
+    FragmentManager fragmentManager = obtainSupportFragmentManager();
+    if (fragmentManager != null) {
+      FeedbackBottomSheet.newInstance(this, FEEDBACK_BOTTOM_SHEET_DURATION)
+              .show(fragmentManager, FeedbackBottomSheet.TAG);
+    }
+  }
+
+  /**
+   * This method enables or disables the alert view from being shown during off-route
+   * events.
+   * <p>
+   * Note this will only happen automatically in the context of
+   * the {@link NavigationView} or a {@link NavigationViewModel}
+   * has been added to the instruction view with {@link InstructionView#subscribe(LifecycleOwner, NavigationViewModel)}.
+   *
+   * @param isEnabled true to show during off-route events, false to hide
+   */
+  public void updateEnabled(boolean isEnabled) {
+    this.isEnabled = isEnabled;
+  }
+
+  @Override
+  public void onFeedbackSelected(FeedbackItem feedbackItem) {
+    if (navigationViewModel == null) {
+      return;
+    }
+//    navigationViewModel.updateFeedback(feedbackItem);// TODO Telemetry impl
```

Comment:
Let's cut tickets with this `TODO` and the rest 👇 so we don't forget 👍

Reply:
https://github.com/mapbox/mapbox-navigation-android/issues/2507